pkg/apis/elasticsearch/v1/elasticsearch_types.go

```diff
@@ -197,7 +197,7 @@ type NodeSet struct {
 	Config *commonv1.Config `json:"config,omitempty"`
 
 	// Count of Elasticsearch nodes to deploy.
-	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Optional
```

Comment:
> I assume to allow nodeSets without count so that autoscaling can set the initial node count?

Yes, exactly. It allows the user to only focus on the min and max count in the autoscaling specification, without the need to initially manage an additional value in the node set. I'll add a comment.